src/note.ts

```diff
@@ -321,95 +546,117 @@ export class Note extends Tickable {
     }
 
     return this.ys;
   }
 
-  // Get the Y position of the space above the stave onto which text can
-  // be rendered.
-  getYForTopText(text_line) {
+  /**
+   * Get the Y position of the space above the stave onto which text can
+   * be rendered.
+   */
+  getYForTopText(text_line: number): number {
     if (!this.stave) {
       throw new Vex.RERR('NoStave', 'No stave attached to this note.');
     }
 
     return this.stave.getYForTopText(text_line);
   }
 
-  // Get a `BoundingBox` for this note.
-  getBoundingBox() {
-    return null;
+  /** Gets a `BoundingBox` for this note. */
+  getBoundingBox(): BoundingBox | undefined {
+    return undefined;
   }
 
-  // Returns the voice that this note belongs in.
-  getVoice() {
+  /** Returns the voice that this note belongs in. */
+  getVoice(): Voice {
     if (!this.voice) throw new Vex.RERR('NoVoice', 'Note has no voice.');
     return this.voice;
   }
 
-  // Attach this note to `voice`.
-  setVoice(voice) {
+  /** Attaches this note to `voice`. */
+  setVoice(voice: Voice): this {
     this.voice = voice;
     this.preFormatted = false;
     return this;
   }
 
-  // Get and set the `TickContext` for this note.
-  getTickContext() {
+  /** Gets the `TickContext` for this note. */
+  getTickContext(): TickContext | undefined {
     return this.tickContext;
   }
-  setTickContext(tc) {
+
+  /** Sets the `TickContext` for this note. */
+  setTickContext(tc: TickContext): this {
     this.tickContext = tc;
     this.preFormatted = false;
     return this;
   }
 
-  // Accessors for the note type.
-  getDuration() {
+  /** Accessors to duration. */
+  getDuration(): string {
     return this.duration;
   }
-  isDotted() {
+
+  /** Accessors to isDotted. */
+  isDotted(): boolean {
     return this.dots > 0;
   }
-  hasStem() {
+
+  /** Accessors to hasStem. */
+  hasStem(): boolean {
     return false;
   }
-  getDots() {
+
+  /** Accessors to dots. */
+  getDots(): number {
     return this.dots;
   }
-  getNoteType() {
+
+  /** Accessors to note type. */
+  getNoteType(): string {
     return this.noteType;
   }
-  setBeam() {
+
+  /** Sets the beam. */
+  setBeam(beam: Beam): this {
+    this.beam = beam;
     return this;
-  } // ignore parameters
+  }
 
-  // Attach this note to a modifier context.
-  setModifierContext(mc) {
+  /** Attach this note to a modifier context. */
+  setModifierContext(mc?: ModifierContext): this {
     this.modifierContext = mc;
     return this;
   }
 
-  // Attach a modifier to this note.
-  addModifier(modifier, index = 0) {
+  /** Attach a modifier to this note. */
+  addModifier(modifier: Modifier, index = 0): this {
     modifier.setNote(this);
     modifier.setIndex(index);
     this.modifiers.push(modifier);
     this.setPreFormatted(false);
     return this;
   }
 
-  // Get the coordinates for where modifiers begin.
-  getModifierStartXY() {
+  /** Get the coordinates for where modifiers begin. */
+  getModifierStartXY(): Coordinates {
```

Comment:
I'd prefer not to have a new type here. `Coordinates` is too generic. Let's use something like:

```
getModifierStartXY(): {x: number, y: number}
```